### PR TITLE
Add filtering to `remote`.

### LINKED ISSUE #128: `remote/2` can filter the host list by role options.
For example:
```
remote :db, primary: true do
  "rm -rf /"
end
```

### PR DESCRIPTION
**Pull request by holetse**
Closes #128

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -296,7 +296,8 @@ end
 
 The workhorse of the Bootleg DSL is `remote`: it executes shell commands on remote servers and returns
 the results. It takes a role and a block of commands to execute. The commands are executed on all servers
-belonging to the role, and raises an `SSHError` if an error is encountered.
+belonging to the role, and raises an `SSHError` if an error is encountered. Optionally, a list of a
```
Comment (rjanja) [inline README.md]: Typo here

File: README.md [hunk 2]
```diff
@@ -321,6 +322,11 @@ end
 remote :app do
   "false"
 end
+
+# filtering - only runs on app hosts with an option of primary set to true
+remote :app, primary: true do
+  "migrate-task"
```
Comment (rjanja) [inline README.md]: Is it obvious that `migrate-task` is a shell command and not something provided by bootleg?

File: README.md (hunk 1 above)
Comment (holetse) [inline README.md]: Fixed!

File: lib/bootleg/config.ex [hunk 3]
```diff
@@ -343,6 +360,10 @@ defmodule Bootleg.Config do
   used as a command. Each command will be simulataneously executed on all hosts in the role. Once
   all hosts have finished executing the command, the next command in the list will be sent.
 
+  `filter` is an optional `Keyword` list of host options to filter with. Any host whose options match
+  the filter will be included in the remote execution. A host matches if it has all of the filtering
+  options defined and the values match (via `==/2`) the filter.
```
Comment (rjanja) [inline lib/bootleg/config.ex]: 👍 for including the comparison used

File: README.md (hunk 2 above)
Comment (holetse) [inline README.md]: Good catch, I'll revise to something more clear.

**Pull request merged**